### PR TITLE
[multistage][testing] Add test for null handling, round func and some other cases

### PR DESCRIPTION
**Pull request by 61yao**

### CODE REVIEW

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-query-runtime/src/test/resources/queries/Cast.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "select_expression_test": {
```
Comment (walterddr) [inline pinot-query-runtime/src/test/resources/queries/Cast.json:2]: Cross merged. Duplicate test case name. Can we hotfix this @61yao ?

Comment (xiangfu0) [reply pinot-query-runtime/src/test/resources/queries/Cast.json:2]: Cast also has issue of:
```
org.apache.pinot.sql.parsers.SqlCompilationException: Caught exception while parsing query: SELECT cast strCol as INT FROM cast_test_cast_as_int

	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileToSqlNodeAndOptions(CalciteSqlParser.java:137)
	at org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:168)
	at org.apache.pinot.query.runtime.QueryRunnerTestBase.queryRunner(QueryRunnerTestBase.java:77)
	at org.apache.pinot.query.runtime.queries.ResourceBasedQueriesTest.runQuery(ResourceBasedQueriesTest.java:178)
	at org.apache.pinot.query.runtime.queries.ResourceBasedQueriesTest.testQueryTestCasesWithH2(ResourceBasedQueriesTest.java:160)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
Caused by: org.apache.pinot.sql.parsers.parser.ParseException: Encountered " "AS" "as "" at line 1, column 20.
Was expecting:
    <EOF> 
    
	at org.apache.pinot.sql.parsers.parser.SqlParserImpl.generateParseException(SqlParserImpl.java:38570)
	at org.apache.pinot.sql.parsers.parser.SqlParserImpl.jj_consume_token(SqlParserImpl.java:38367)
	at org.apache.pinot.sql.parsers.parser.SqlParserImpl.SqlStmtsEof(SqlParserImpl.java:1031)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.compileToSqlNodeAndOptions(CalciteSqlParser.java:127)
	... 28 more
```

File: pinot-query-runtime/src/test/resources/queries/NullHanlding.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "select_expression_test": {
```
Comment (xiangfu0) [inline pinot-query-runtime/src/test/resources/queries/NullHanlding.json:2]: duplicated test name.

Comment (xiangfu0) [inline pinot-query-runtime/src/test/resources/queries/NullHanlding.json:2]: Also seeing this error after changed the test name:
```

java.lang.AssertionError: ANY

	at org.apache.calcite.sql.type.SqlTypeExplicitPrecedenceList.compareTypePrecedence(SqlTypeExplicitPrecedenceList.java:141)
	at org.apache.calcite.sql.SqlUtil.bestMatch(SqlUtil.java:775)
	at org.apache.calcite.sql.SqlUtil.filterRoutinesByTypePrecedence(SqlUtil.java:730)
	at org.apache.calcite.sql.SqlUtil.lookupSubjectRoutines(SqlUtil.java:566)
	at org.apache.calcite.sql.SqlUtil.lookupRoutine(SqlUtil.java:486)
	at org.apache.calcite.sql.SqlFunction.deriveType(SqlFunction.java:253)
	at org.apache.calcite.sql.SqlFunction.deriveType(SqlFunction.java:231)
	at org.apache.calcite.sql.validate.SqlValidatorImpl$DeriveTypeVisitor.visit(SqlValidatorImpl.java:6277)
	at org.apache.calcite.sql.validate.SqlValidatorImpl$DeriveTypeVisitor.visit(SqlValidatorImpl.java:6264)
	at org.apache.calcite.sql.SqlCall.accept(SqlCall.java:161)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.deriveTypeImpl(SqlValidatorImpl.java:1862)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.deriveType(SqlValidatorImpl.java:1847)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.expandSelectItem(SqlValidatorImpl.java:463)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateSelectList(SqlValidatorImpl.java:4409)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateSelect(SqlValidatorImpl.java:3652)
	at org.apache.calcite.sql.validate.SelectNamespace.validateImpl(SelectNamespace.java:64)
	at org.apache.calcite.sql.validate.AbstractNamespace.validate(AbstractNamespace.java:89)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateNamespace(SqlValidatorImpl.java:1100)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateQuery(SqlValidatorImpl.java:1071)
	at org.apache.calcite.sql.SqlSelect.validate(SqlSelect.java:247)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateScopedExpression(SqlValidatorImpl.java:1046)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validate(SqlValidatorImpl.java:752)
	at org.apache.pinot.query.QueryEnvironment.validate(QueryEnvironment.java:192)
	at org.apache.pinot.query.QueryEnvironment.compileQuery(QueryEnvironment.java:183)
	at org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:131)
	at org.apache.pinot.query.QueryEnvironment.planQuery(QueryEnvironment.java:168)
	at org.apache.pinot.query.runtime.QueryRunnerTestBase.queryRunner(QueryRunnerTestBase.java:77)
	at org.apache.pinot.query.runtime.queries.ResourceBasedQueriesTest.runQuery(ResourceBasedQueriesTest.java:178)
	at org.apache.pinot.query.runtime.queries.ResourceBasedQueriesTest.testQueryTestCasesWithH2(ResourceBasedQueriesTest.java:160)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
```

**Pull request merged**